### DISCUSSION TITLE
Could OpenPNM simulate the seepage through materials with different porous structure?

### DISCUSSION THREAD

**Discussion by GaoC-en** [PMEAL/OpenPNM#2360] [Q&A]
Dear professor:
I am really fond of OpenPNM, it is awesome! But I want to know could the program simulate the seepage through materials with different porous structure? For example, the water flow through the fine-grained sand into coarse-grained, maybe the permeability _K_ of these two materials changes from E-4 to E-1. I do not know how to simulate the throats change at the interface. Thanks a lot for your help!

**Comment by jgostick** [comment]
This is possible for sure...in fact the idea of having 'multiple layers' is at the core of OpenPNM's design, and actually causes quite a bit of headache, but we keep it around for situations like this.  Basically, the steps are: 

* Make 2 networks, one coarse and one fine, with similar x-y dimensions but (maybe) different thicknesses in the z direction
* Adjust the a coordinates of the 'coarse' network so that it is positioned above the fine one (or vice versa)
* You can them 'stitch' these two networks together, which not only combines the coords and conns of the two networks into the same object, but also creates throat connections between them.
* Now, you can calculate the geometry of the pores and throats for each of the layers, which is where OpenPNM's objects come into play.
* You would create one geometry objects for the fine layer, and another for the coarse layer, and 'maybe' another for the throats that stitch the two layers together.  These throats could potential be part of the coarse geometry as well.  Anyway, each of these geometry objects will have it's own pores size distribution model, which is why you need 2 objects.

There is an example notebook [here](https://github.com/PMEAL/OpenPNM/blob/dev/examples/tutorials/network/stitching_and_merging_networks_together.ipynb) that outlines the above steps.

**Reply by GaoC-en** [reply]
Thanks for your patient reply! I will try it later
